crates/mitosis/evm/src/executor.rs

```diff
@@ -0,0 +1,120 @@
+use alloy_evm::{
+    block::{
+        BlockExecutionError, BlockExecutionResult, BlockExecutor,
+        BlockExecutorFactory, BlockExecutorFor, CommitChanges, 
+        ExecutableTx, OnStateHook,
+    },
+    Database, Evm, EvmFactory,
+};
+use alloy_primitives::Log;
+use alloy_consensus::{Transaction, TxReceipt};
+use alloy_eips::eip2718::Encodable2718;
+use revm::{context::result::ExecutionResult, Inspector, DatabaseCommit};
+use crate::system_calls::apply_multicall3_deployment;
+
+/// Wrapper block executor for Mitosis that delegates to an inner executor
+/// while maintaining full compatibility with existing functionality.
+#[derive(Debug)]
+pub struct MitosisBlockExecutor<Inner> {
+    inner: Inner,
+}
+
+impl<Inner> MitosisBlockExecutor<Inner> {
+    /// Creates a new MitosisBlockExecutor wrapping the given executor.
+    pub fn new(inner: Inner) -> Self {
+        Self { inner }
+    }
+}
+
+impl<Inner> BlockExecutor for MitosisBlockExecutor<Inner>
+where
+    Inner: BlockExecutor,
+    Inner::Transaction: Transaction + Encodable2718,
+    Inner::Receipt: TxReceipt<Log = Log>,
+    <Inner::Evm as Evm>::DB: Database + DatabaseCommit,
+{
+    type Transaction = Inner::Transaction;
+    type Receipt = Inner::Receipt;
+    type Evm = Inner::Evm;
+
+    fn apply_pre_execution_changes(&mut self) -> Result<(), BlockExecutionError> {
+        // First apply the inner pre-execution changes
+
+        self.inner.apply_pre_execution_changes()?;
+        
+        // Deploy Multicall3 contract using system transaction at block 150
+        apply_multicall3_deployment(self.inner.evm_mut())?;
```

Comment:
I think we should apply this only for mainnet. We should not run this logic if it is not a mainnet.

Reply:
https://github.com/mitosis-org/reth/pull/1/files#diff-bcfae8e1e0afb3cc7090746b8ecf7d35fd2d3aa08297632b8a2f4c2de7d78211R28-R40

I fix it. we should change this feature to real hardfork function when we ready.